### PR TITLE
Dependencies fix

### PR DESCRIPTION
**Pull request by 3asm**
Make the google-cloud-logging dep optional and handle error if it is missing. Ping urllib2 and remove pin on requests.
Handle docker API error.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -50,8 +50,9 @@ install_requires =
     docker
     importlib-metadata; python_version<"3.8"
     jsonschema>=4.4.0
-    protobuf==3.20.1
-    requests<=2.28.1
+    protobuf~=3.20
+    urllib3<2
+    requests
```
Comment (elyousfi5) [inline setup.cfg:55]: did we check if this wont create an issue any more ?

File: src/ostorlab/runtimes/local/runtime.py [hunk 2]
```diff
@@ -532,28 +533,30 @@ def list(self, page: int = 1, number_elements: int = 10) -> List[runtime.Scan]:
...
-                        console.warning(
-                            f"Scan {ostorlab_universe_id} has not traced in DB."
-                        )
-            except KeyError:
-                logger.warning("The label ostorlab.universe do not exist.")
-
-        return list(scans.values())
+                        universe_ids.add(ostorlab_universe_id)
+                        if (
+                            ostorlab_universe_id.isnumeric()
```
Comment (elyousfi5) [inline src/ostorlab/runtimes/local/runtime.py:548]:
```suggestion
                            ostorlab_universe_id.isnumeric() is True
```

File: src/ostorlab/cli/rootcli.py [hunk 3]
```diff
@@ -61,11 +58,19 @@ def rootcli(
...
-        client = google.cloud.logging.Client(credentials=credentials)
-        client.setup_logging()
-        ctx.obj["gcp_logging_credential"] = content
+            credentials = service_account.Credentials.from_service_account_info(info)
+            client = google.cloud.logging.Client(credentials=credentials)
+            client.setup_logging()
+            ctx.obj["gcp_logging_credential"] = content
+        except ImportError:
+            logger.error(
+                "Could not import Google Cloud Logging, install it with `pip install 'ostorlab[google-cloud-logging]'"
```
Comment (benyissa) [inline src/ostorlab/cli/rootcli.py:75]: Why not installing google-cloud-logging directly ?

File: setup.cfg (hunk 1 above)
Comment (amine3) [reply setup.cfg:55]: +1 We need to stick to that version until we test and validate the latest version

**Review by amine3 — CHANGES_REQUESTED** [review]

File: src/ostorlab/cli/rootcli.py [hunk 4]
```diff
@@ -61,11 +58,19 @@ def rootcli(
...
 
-        credentials = service_account.Credentials.from_service_account_info(info)
-        client = google.cloud.logging.Client(credentials=credentials)
-        client.setup_logging()
-        ctx.obj["gcp_logging_credential"] = content
+            credentials = service_account.Credentials.from_service_account_info(info)
+            client = google.cloud.logging.Client(credentials=credentials)
+            client.setup_logging()
+            ctx.obj["gcp_logging_credential"] = content
+        except ImportError:
```
Comment (oussamaessaji) [inline src/ostorlab/cli/rootcli.py:73]: Would it be possible to make this step optional by passing an argument to the CLI?

**Review by mohsinenar — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (3asm) [reply setup.cfg:55]: I did test it.

Comment (3asm) [reply setup.cfg:55]: Plus the `requests` has documentation on this.

File: src/ostorlab/cli/rootcli.py (hunk 3 above)
Comment (3asm) [reply src/ostorlab/cli/rootcli.py:75]: Too big.

**Review by amine3 — APPROVED** [review]

**Review by PiranhaSa — APPROVED** [review]

**Pull request merged**